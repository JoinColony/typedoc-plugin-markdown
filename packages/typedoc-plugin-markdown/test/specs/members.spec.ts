import * as Handlebars from 'handlebars/runtime';
import { ProjectReflection } from 'typedoc';

describe(`Members:`, () => {
  let membersPartial: Handlebars.TemplateDelegate;
  let memberPartial: Handlebars.TemplateDelegate;
  let project: ProjectReflection;

  describe(`(members)`, () => {
    beforeAll(async () => {
      const bootstrap = global.bootstrap('members.ts', {
        stubPartials: ['member', 'member.sources'],
      });
      project = bootstrap.project;
      membersPartial = global.getTemplate(bootstrap.context, 'members');
      memberPartial = global.getTemplate(bootstrap.context, 'member');
    });
    test(`should compile module members'`, () => {
      expect(
        global.renderTemplate(
          membersPartial,
          global.findModule(project, 'members'),
        ),
      ).toMatchSnapshot();
    });

    test(`should compile class members'`, () => {
      expect(
        global.renderTemplate(
          membersPartial,
          project.findReflectionByName('ClassWithAccessorMembers'),
        ),
      ).toMatchSnapshot();
    });
  });

  describe(`(member)`, () => {
    beforeAll(async () => {
      const bootstrap = global.bootstrap('members.ts', {
        stubPartials: ['member.sources'],
      });
      project = bootstrap.project;
      membersPartial = global.getTemplate(bootstrap.context, 'members');
      memberPartial = global.getTemplate(bootstrap.context, 'member');
    });
    test(`should compile declaration members'`, () => {
      expect(
        global.renderTemplate(
          memberPartial,
          project.findReflectionByName('declarationMember'),
        ),
      ).toMatchSnapshot();
    });

    test(`should compile declaration member with named anchors'`, () => {
      expect(
        global.renderTemplate(
          memberPartial,
          project.findReflectionByName('declarationMember'),
          { namedAnchors: true },
        ),
      ).toMatchSnapshot();
    });

    test(`should compile a signature members'`, () => {
      expect(
        global.renderTemplate(
          memberPartial,
          project.findReflectionByName('signatureMember'),
        ),
      ).toMatchSnapshot();
    });

    test(`should compile members with getter'`, () => {
      expect(
        global.renderTemplate(
          memberPartial,
          (
            project.findReflectionByName('ClassWithAccessorMembers') as any
          ).findReflectionByName('getter'),
        ),
      ).toMatchSnapshot();
    });

    test(`should compile members with setter'`, () => {
      expect(
        global.renderTemplate(
          memberPartial,
          (
            project.findReflectionByName('ClassWithAccessorMembers') as any
          ).findReflectionByName('setter'),
        ),
      ).toMatchSnapshot();
    });
  });
});
