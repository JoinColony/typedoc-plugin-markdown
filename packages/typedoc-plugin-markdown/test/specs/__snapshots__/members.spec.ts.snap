// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Members: (members) should compile class members' 1`] = `
"## Constructors

### constructor

• **new ClassWithAccessorMembers**()

[sources]

## Properties

### \\\\_private

• \`Private\` **\\\\_private**: \`string\`

[sources]

## Accessors

### getter

• \`get\` **getter**(): \`string\`

#### Returns

\`string\`

[sources]

---

### setter

• \`set\` **setter**(\`value\`): \`void\`

#### Parameters

| Name | Type |
| :------ | :------ |
| \`value\` | \`string\` |

#### Returns

\`void\`

[sources]
"
`;
